### PR TITLE
new onBoarding Modal

### PR DESCRIPTION
**Pull request by Ladvace**

### CODE REVIEW

**Comment by killpowa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/common/modals/Onboarding/index.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/gorilla-devs/GDLauncher/pullRequest?prid=5328128&bid=17439995)

File: src/app/desktop/views/Home.js [hunk 1]
```diff
@@ -28,18 +28,29 @@ const Home = () => {
...
   const account = useSelector(_getCurrentAccount);
   const news = useSelector(state => state.news);
+  const isNewUser = useSelector(state => state.app.isNewUser);
 
   const openAddInstanceModal = defaultPage => {
     dispatch(openModal('AddInstance', { defaultPage }));
   };
 
+  const openWelcomePage = () => {
+    console.log('NEW', isNewUser);
```
Comment (blarfoon) [inline src/app/desktop/views/Home.js]: Remove

File: src/app/desktop/views/Home.js [hunk 2]
```diff
@@ -28,18 +28,29 @@ const Home = () => {
...
+
   const openAccountModal = () => {
     dispatch(openModal('AccountsManager'));
   };
   const [profileImage, setProfileImage] = useState(null);
 
   useEffect(() => {
     extractFace(account.skin).then(setProfileImage).catch(console.error);
+
+    console.log('K', isNewUser);
```
Comment (blarfoon) [inline src/app/desktop/views/Home.js]: Remove

File: src/app/desktop/views/Home.js [hunk 3]
```diff
@@ -28,18 +28,29 @@ const Home = () => {
...
+    if (isNewUser) {
+      openWelcomePage();
```
Comment (blarfoon) [inline src/app/desktop/views/Home.js]: It's a modal not a page

File: src/common/modals/Onboarding/index.js [hunk 4]
```diff
@@ -1,100 +1,226 @@
...
+        <SideBar>
+          {slides.map((title, i) => {
+            return (
+              <SectionButton
+                // active={page === 1}
+                onClick={() => {
+                  console.log('£££', i, page);
+                  setPage(i);
+                  if (i > page) {
+                    console.log('pippo ', page + i);
```
Comment (blarfoon) [inline src/common/modals/Onboarding/index.js]: Pippo

File: src/common/modals/Onboarding/index.js [hunk 5]
```diff
@@ -1,100 +1,226 @@
...
+      removePadding
+    >
+      <Container>
+        <SideBar>
+          {slides.map((title, i) => {
+            return (
+              <SectionButton
+                // active={page === 1}
+                onClick={() => {
+                  console.log('£££', i, page);
```
Comment (blarfoon) [inline src/common/modals/Onboarding/index.js]: ...

File: src/common/reducers/actions.js [hunk 6]
```diff
@@ -437,7 +434,6 @@ export function loginThroughNativeLauncher() {
 
       if (isNewUser) {
         dispatch(updateIsNewUser(false));
-        dispatch(push('/onboarding'));
```
Comment (blarfoon) [inline src/common/reducers/actions.js:453]: If it's a new user they won't be redirected anwhere

File: src/common/reducers/actions.js [hunk 7]
```diff
@@ -341,9 +341,6 @@ export function login(username, password, redirect = true) {
         }
       } else {
         dispatch(updateIsNewUser(false));
-        if (redirect) {
-          dispatch(push('/onboarding'));
```
Comment (blarfoon) [inline src/common/reducers/actions.js:354]: If it's a new user they won't be redirected anwhere

File: yarn.lock [hunk 8]
```diff
@@ -4994,7 +4994,7 @@ debug@=3.1.0:
   dependencies:
     ms "2.0.0"
 
-debug@^3.0.0, debug@^3.1.1, debug@^3.2.5, debug@^3.2.6:
```
Comment (blarfoon) [inline yarn.lock]: Why change the yarn lock?

File: src/common/modals/Onboarding/index.js [hunk 9]
```diff
@@ -1,100 +1,226 @@
...
+const SideBar = styled.div`
+  left: 0;
+  flex: 0;
+  flex-grow: 1;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-evenly;
+  align-items: flex-end;
+  background: ${props => props.theme.palette.grey[800]};
+  padding-top: calc(${props => props.theme.sizes.height.systemNavbar} + 5px);
```
Comment (blarfoon) [inline src/common/modals/Onboarding/index.js]: This doesn't look solid. Why `props.theme.sizes.height.systemNavbar} + 5px`?

File: src/common/modals/Onboarding/index.js [hunk 10]
```diff
@@ -1,100 +1,226 @@
...
+import Modal from '../../components/Modal';
+import { _getCurrentAccount } from '../../utils/selectors';
+import logo from '../../assets/logo.png';
+
+const scrollToRef = ref =>
+  ref.current.scrollIntoView({ behavior: 'smooth', block: 'center' });
+
+const Onboarding = () => {
+  const [page, setPage] = useState(0);
+  const [currentSlide, setCurrentSlide] = useState(0);
```
Comment (blarfoon) [inline src/common/modals/Onboarding/index.js:14]: What's the difference between page and currentSlide?

**Review by blarfoon — CHANGES_REQUESTED** [review]

**Comment by killpowa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/common/modals/Onboarding/index.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/gorilla-devs/GDLauncher/pullRequest?prid=5328128&bid=17439995)

File: src/common/reducers/actions.js [hunk 11]
```diff
@@ -340,6 +340,9 @@ export function login(username, password, redirect = true) {
         }
       } else {
         dispatch(updateIsNewUser(false));
+        if (redirect) {
+          dispatch(push('/onboarding'));
+        }
       }
```
Comment (blarfoon) [inline src/common/reducers/actions.js]: Onboarding doesn't exist anymore, this will fail

File: src/common/reducers/actions.js [hunk 12]
```diff
@@ -433,6 +436,7 @@ export function loginThroughNativeLauncher() {
 
       if (isNewUser) {
         dispatch(updateIsNewUser(false));
+        dispatch(push('/onboarding'));
```
Comment (blarfoon) [inline src/common/reducers/actions.js]: Same as above

**Review by blarfoon — CHANGES_REQUESTED** [review]

File: src/app/desktop/views/Home.js [hunk 13]
```diff
@@ -28,18 +28,26 @@ const Home = () => {
...
+  const isNewUser = useSelector(state => state.app.isNewUser);
 
   const openAddInstanceModal = defaultPage => {
     dispatch(openModal('AddInstance', { defaultPage }));
   };
 
+  const openWelcomeModal = () => {
+    dispatch(openModal('Onboarding'));
+  };
+
```
Comment (blarfoon) [inline src/app/desktop/views/Home.js]: What's the point of this function? It's used in only 1 place and it does only one thing

File: src/common/modals/Onboarding/index.js [hunk 14]
```diff
@@ -1,100 +1,224 @@
...
+        <SideBar>
+          {slides.map((title, i) => {
+            return (
+              <SectionButton
+                // active={page === 1}
+                onClick={() => {
+                  setPage(i);
+                  if (i > page) {
+                    executeScroll(i - page);
+                  } else executeScroll(i - page);
```
Comment (blarfoon) [inline src/common/modals/Onboarding/index.js]: ??

File: src/app/desktop/views/Home.js [hunk 15]
```diff
@@ -28,18 +28,26 @@ const Home = () => {
...
   const openAccountModal = () => {
     dispatch(openModal('AccountsManager'));
   };
   const [profileImage, setProfileImage] = useState(null);
 
   useEffect(() => {
     extractFace(account.skin).then(setProfileImage).catch(console.error);
+    if (isNewUser) {
+      openWelcomeModal();
+    }
```
Comment (blarfoon) [inline src/app/desktop/views/Home.js]: This should not be here. This should be called from the action

**Review by blarfoon — CHANGES_REQUESTED** [review]
Pr is not complete. There are still references to `/onboarding` and there are some blank redirects

**Comment by killpowa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/common/modals/Onboarding/index.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/gorilla-devs/GDLauncher/pullRequest?prid=5328128&bid=17439995)

**Comment by killpowa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/common/modals/Onboarding/index.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/gorilla-devs/GDLauncher/pullRequest?prid=5328128&bid=17439995)

**Comment by blarfoon** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- src/app/desktop/views/Onboarding.js  4
- src/common/modals/Onboarding/index.js  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/gorilla-devs/GDLauncher/pullRequest?prid=5328128&bid=17439995)

**Pull request closed without merging**